### PR TITLE
UIF-111 Search and Filter "Group"

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
When searching for a specific type of finance record the user must be able to search and filter by specific criteria. In order to find the record they need.

Filters 
- Status
- Acquisition units

Search
- Keyword (search the name and description fields)
- Name
- Code

https://issues.folio.org/browse/UIF-111
## Approach
Rewrite functional component `GroupsList` into a class component for using common functions for filters and search.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/Groups/GroupsList/GroupsList.js [hunk 1]
```diff
@@ -40,92 +57,136 @@ const columnMapping = {
...
+    }
+  }
 
-GroupsList.propTypes = {
-  mutator: PropTypes.object.isRequired,
-  resources: PropTypes.object.isRequired,
-  stripes: PropTypes.object.isRequired,
-  history: ReactRouterPropTypes.history.isRequired,
-};
+  onChangeIndex = (e) => {
```
Comment (NikitaSedyx) [inline src/Groups/GroupsList/GroupsList.js]: it's not used

File: src/Groups/GroupsList/GroupsList.js [hunk 2]
```diff
@@ -40,92 +57,130 @@ const columnMapping = {
...
+    },
+    resultCount: { initialValue: INITIAL_RESULT_COUNT },
+    records: {
+      type: 'okapi',
+      clear: true,
+      records: 'groups',
+      recordsRequired: '%{resultCount}',
+      path: GROUPS_API,
+      perRequest: RESULT_COUNT_INCREMENT,
+      throwErrors: false,
```
Comment (aliaksei-chumakou) [inline src/Groups/GroupsList/GroupsList.js]: don't we have some `baeResource` in acq-components?

File: src/Groups/GroupsList/GroupsList.js [hunk 3]
```diff
@@ -40,92 +57,130 @@ const columnMapping = {
...
 
-  const onCreate = useCallback(
-    async (group) => {
-      try {
-        const savedGroup = await mutator.records.POST(group);
+    this.getActiveFilters = getActiveFilters.bind(this);
+    this.handleFilterChange = handleFilterChange.bind(this);
+    this.changeSearchIndex = changeSearchIndex.bind(this);
+    this.callout = React.createRef();
+    this.showToast = showToast.bind(this);
```
Comment (aliaksei-chumakou) [inline src/Groups/GroupsList/GroupsList.js]: don't we added this to index and use a common hook?

File: src/Groups/GroupsList/GroupsList.js [hunk 4]
```diff
@@ -40,92 +57,130 @@ const columnMapping = {
...
+          parentMutator={mutator}
+          stripes={stripes}
+          onFilterChange={this.handleFilterChange}
+          renderFilters={this.renderFilters}
+          renderNavigation={this.renderNavigation}
+          searchableIndexes={this.getTranslateSearchableIndexes()}
+          selectedIndex={get(resources.query, 'qindex')}
+          onChangeIndex={this.changeSearchIndex}
+        />
+        <Callout ref={this.callout} />
```
Comment (aliaksei-chumakou) [inline src/Groups/GroupsList/GroupsList.js]: the same question, isn't it a hook?

File: src/Groups/GroupsListFilters/GroupsListFiltersContainer.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+
+import { acqUnitsResource } from '../../common/resources';
+
```
Comment (aliaksei-chumakou) [inline src/Groups/GroupsListFilters/GroupsListFiltersContainer.js]: extra line

File: src/Groups/GroupsList/GroupsList.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/Groups/GroupsList/GroupsList.js]: But we use hooks for functional components

File: src/Groups/GroupsListFilters/GroupsListFiltersContainer.js (hunk 5 above)
Comment (AndreiShumski2) [reply src/Groups/GroupsListFilters/GroupsListFiltersContainer.js]: Removed. thnx

File: src/Groups/GroupsList/GroupsList.js (hunk 3 above)
Comment (aliaksei-chumakou) [reply src/Groups/GroupsList/GroupsList.js]: yes, you right, let's keep it the olds style till we switch on functional component list

File: src/Groups/GroupsList/GroupsList.js [hunk 6]
```diff
@@ -40,92 +58,129 @@ const columnMapping = {
...
-        filterConfig={FILTER_CONFIG}
-        renderNavigation={renderNavigation}
-      />
-    </div>
-  );
-};
+      return savedGroup;
+    } catch (response) {
+      showToast('ui-finance.groups.actions.save.error', 'error');
+      this.showToast('ui-finance.groups.actions.save.error', 'error');
```
Comment (aliaksei-chumakou) [inline src/Groups/GroupsList/GroupsList.js]: whaaat?) twice, for extra guarantee?)

Comment (AndreiShumski2) [reply src/Groups/GroupsList/GroupsList.js]: Sure 😊. Removed it.

**Pull request merged**